### PR TITLE
feat: add rustc + cargo rustc spec

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature
**What is the current behavior? (You can also link to an open issue here)**
#292 No support for rustc or cargo rustc
**What is the new behavior (if this is a feature change)?**
Adds support for rustc and cargo rustc
**Additional info:**

### CODE REVIEW

File: dev/cargo.ts [hunk 1]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+                    },
+                    {
+                      name: "KIND=framework",
+                      description:
+                        "Only search for macOS frameworks in this directory",
+                    },
+                    {
+                      name: "KIND=all",
+                      description:
+                        "Search for all library kinds in this directory",
```
Comment (QuiiBz) [inline dev/cargo.ts:115]: I think it's hard to understand the name of the suggestions with this `KIND=`. Let's simply write `dependency`, `crate`, `native`...

File: dev/cargo.ts [hunk 2]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+                    },
+                  ],
+                  isOptional: true,
+                },
+                {
+                  name: "path",
+                  description:
+                    "Path to search for external crates and libraries",
+                },
```
Comment (QuiiBz) [inline dev/cargo.ts:124]: Looks like this can have a template of folders?

File: dev/cargo.ts [hunk 3]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+                      description: "A native dynamic library",
+                    },
+                    {
+                      name: "KIND=static",
+                      description: "A native static library",
+                    },
+                    {
+                      name: "KIND=framework",
+                      description: "A macOS framework",
+                    },
```
Comment (QuiiBz) [inline dev/cargo.ts:146]: Same as above

File: dev/cargo.ts [hunk 4]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+                suggestions: ["auto", "always", "never"],
+              },
+            },
+          ],
+        },
+      ],
+      args: {
+        template: "filepaths",
+        name: "file",
+        description: "Additional files to compile",
```
Comment (QuiiBz) [inline dev/cargo.ts:433]: Based on this description, it seems that this arg is variadic

File: dev/cargo.ts [hunk 5]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+        },
+        {
+          name: "--example",
+          description: "Builds only the specified example",
+          args: {
+            name: "Name",
+          },
+        },
+        {
+          name: "--examples  ",
```
Comment (QuiiBz) [inline dev/cargo.ts:479]: 2 extra spaces

File: dev/cargo.ts [hunk 6]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+          args: {
+            name: "Info",
+          },
+        },
+        {
+          name: "--target-dir",
+          description: "Directory for all generated artifacts",
+          args: {
+            name: "Directory",
+          },
```
Comment (QuiiBz) [inline dev/cargo.ts:541]: Can this arg have a template of `folders`?

File: dev/cargo.ts [hunk 7]
```diff
@@ -60,6 +60,582 @@ export const completionSpec: Fig.Spec = {
...
+        },
+        {
+          name: "--manifest-path",
+          description: "Path to Cargo.toml",
+          args: {
+            name: "Path",
+          },
```
Comment (QuiiBz) [inline dev/cargo.ts:548]: Based on the description, this arg can have a template of `filepaths`

File: dev/rustc.ts [hunk 8]
```diff
@@ -0,0 +1,339 @@
+export const completion: Fig.Spec = {
+  name: "rustc",
+  description: "CLI for rustc",
+  subcommands: [],
```
Comment (QuiiBz) [inline dev/rustc.ts:4]: No need to set an empty array if this command doesn't have any subcommands

File: dev/rustc.ts [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+              description: "Only search for native libraries in this directory",
+            },
+            {
+              name: "KIND=framework",
+              description: "Only search for macOS frameworks in this directory",
+            },
+            {
+              name: "KIND=all",
+              description: "Search for all library kinds in this directory",
+            },
```
Comment (QuiiBz) [inline dev/rustc.ts:52]: Same as above for this `KIND=` thing

File: dev/rustc.ts [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+      name: ["-v", "--verbose"],
+      description: "Use verbose output",
+    },
+    {
+      name: "--extern",
+      description: "Specify where an external library is located",
+      args: {
+        name: "Path",
+        description: "Path where crate can be found",
+      },
```
Comment (QuiiBz) [inline dev/rustc.ts:312]: Looks like this arg can have a template of `folders`

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Rustaceans will be very happy to have those specs!

File: dev/cargo.ts (hunk 1 above)
Comment (Monkeyank) [reply dev/cargo.ts:115]: The required argument value provided has to be `KIND="type"`, so instead should I change the name to `dependency`, `crate`, `native` ... . Then, add the insert value prop, and add the full value, ex: `KIND=crate`?

Comment (QuiiBz) [reply dev/cargo.ts:115]: Agree, that would be better for the users to have a clear and concise name. Let's do this!

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**